### PR TITLE
Use French-style tree rendering

### LINKED ISSUE #138: Better Rendering for Trees
Right now trees look totally strange in Openstreetmap:
before

I propose a style like this: its basically a cycle with the same pattern that `landuse=forrest` uses, rendered on the same layer as `landuse=forrest`.

after

That causes trees being invisible *inside* forests and close to it trees just *melts* with it. what do you think?

### LINKED ISSUE #838: Trees rendered too early
It seems that trees are currently rendered too early. [Example](http://www.openstreetmap.org/#map=17/49.62129/6.15032)

Perhaps we should move tree rendering to z18, except for trees that have a name (and therefore thus can be seen as landmarks).

### PR DESCRIPTION
**Pull request by math1985**
Merge tree rendering style developed by @cquest for French rendering, see
https://github.com/cquest/osmfr-cartocss/.

* Resolves #138
* Resolves #838

### CODE REVIEW

**Comment by math1985** [comment]
@gravitystorm This still needs a decision which of both options to go for, see https://github.com/gravitystorm/openstreetmap-carto/pull/978#issuecomment-61204056. If you could let us know which option you prefer I can update my PR.

**Comment by gravitystorm** [comment]
Hmm. I don't have a particularly strong opinion on this - there's clearly tradeoffs either way. I don't like the SQL being so complicated - it took me a while to figure out what exactly is going on with the st_collect for example (and even then, I had to go back and reconsider a couple of times). So I think the CartoCSS is easier to figure out than the SQL, for non-postgis experts.

Also, I don't like the blur. We've moved to very sharp icons and it seems unusual that we'd blur the trees.

**Comment by math1985** [comment]
Rebased, merged @pnorman's proposal, and removed blur.

**Comment by math1985** [comment]
Before:
screenshot from 2015-01-30 21 48 19




After:
screenshot from 2015-01-30 21 47 14
screenshot from 2015-01-30 21 47 01
screenshot from 2015-01-30 21 46 46

**Comment by HolgerJeromin** [comment]
Am i right that you want to push this into master and not adding names of trees right now? #801

**Comment by math1985** [comment]
Yes, I think the too prominent rendering is a bigger problem than missing names, so I'd like to have this merged first.

File: project.yaml [hunk 1]
```diff
@@ -1195,6 +1181,53 @@ Layer:
                  OR aerialway = 'station'
               ) AS stations_poly
     advanced: {}
+  - id: "glaciers-text"
```
Comment (pnorman) [inline project.yaml]: is this a rebase gone wrong?

**Comment by imagico** [comment]
Is there a reason why width for tree_row and tree are different on z=19/20?

The transparency in rendering will lead to differences depending on the background color - this might make it difficult to recognize and it might look very faint on green landcovers - testing also in a rural environment would probably be a good idea - like [here](http://www.openstreetmap.org/#map=16/47.8230/8.0721).

File: project.yaml (hunk 1 above)
Comment (math1985) [inline project.yaml]: Thanks for catching, fixed.

**Comment by math1985** [comment]
> Is there a reason why width for tree_row and tree are different on z=19/20?

I don't think so, I made them the same now.

**Comment by math1985** [comment]
> The transparency in rendering will lead to differences depending on the background color - this might make it difficult to recognize and it might look very faint on green landcovers - testing also in a rural environment would probably be a good idea - like here.

Seems to work reasonably well on different backgrounds:

screenshot from 2015-01-31 00 36 10
screenshot from 2015-01-31 00 36 08
screenshot from 2015-01-31 00 35 54

**Comment by imagico** [comment]
Colors seem fine.

Regarding scaling - here for reference the actual real world sizes of the width values used:

zoom|pixel size (m)|width|real width at equator (m)|real width at 70° lat (m)
----|--------------|-----|-------------------------|-------------------------
16 | 2.39 | 2.5 | 6 | 2
17 | 1.19 | 5 | 6 | 2
18 | 0.6 | 10 | 6 | 2
19 | 0.3 | 15 | 4.5 | 1.5
20 | 0.15 | 30 | 4.5 | 1.5

This might be a good opportunity to try out dynamic scaling depending on latitude - with something like

`SELECT way, "natural", cosh(0.5*(ST_YMin(way)+ST_YMax(way))/6378137.0) as scale_factor`

you can adjust the width to the varying map scale.

**Comment by math1985** [comment]
I'm not sure if dynamic scaling for some icons/objects but not others would work well. In this case I don't think it would work well.

**Comment by math1985** [comment]
Why does Github think this cannot be automatically merged? It has been rebased, and locally a merge seems to go fine.

**Comment by math1985** [comment]
> Why does Github think this cannot be automatically merged? It has been rebased on master, and locally a merge seems to go fine.

Fixed.

**Comment by imagico** [comment]
> I'm not sure if dynamic scaling for some icons/objects but not others would work well. In this case I don't think it would work well.

That is possible - will review it once it is live on the map.

**Pull request merged**